### PR TITLE
Add docker notebook configuration

### LINKED ISSUE #151: Add docker-compose-notebook.yml
https://github.com/mitodl/mitxpro/pull/292

### LINKED ISSUE #185: Update README
We should reference https://github.com/mitodl/handbook/blob/master/common-web-app-guide.md similar to how the xPRO README does

### PR DESCRIPTION
**Pull request by noisecapella**
Fixes #151 
Fixes #185

### CODE REVIEW

**Comment by noisecapella** [comment]
@mbertrand Made the change. Can you go through the steps and confirm it works? Some of the instructions are for OS X only

**Comment by noisecapella** [comment]
@mbertrand Also, I made some more drastic changes to the README to cut outdated stuff and to try to match xPRO

File: {{ cookiecutter.project_name }}/README.md [hunk 1]
```diff
@@ -113,3 +113,45 @@ Tests should be run in the Docker container, not the host machine. They can be r
...
+    notebook_1  |     To access the notebook, open this file in a browser:
+    notebook_1  |         file:///home/mitodl/.local/share/jupyter/runtime/nbserver-8-open.html
+    notebook_1  |     Or copy and paste one of these URLs:
+    notebook_1  |         http://(2c19429d04d0 or 127.0.0.1):8080/?token=2566e5cbcd723e47bdb1b058398d6bb9fbf7a31397e752ea
+    ```
+  Here is a one-line command that will produce a browser-ready URL from that output. Run this in a separate terminal:
+    ```bash
+    docker logs $(docker ps --format '{{.Names}}' | grep "_notebook_run_") | grep -E "http://(.*):8080[^ ]+\w" | tail -1 | sed -e 's/^[[:space:]]*//'
+    ```
+  OSX users can pipe that output to `xargs open` to open a browser window directly with the URL from that command.
```
Comment (mbertrand) [inline {{ cookiecutter.project_name }}/README.md:88]: I had to modify this to work on OSX:
```bash
docker logs $(docker ps --format '{{ .Names}}' | grep "_notebook_run_") | grep -E "http://(.*):8080[^ ]+\w" | tail -1 | sed -e 's/^[[:space:] or]*//'
```

File: {{ cookiecutter.project_name }}/README.md [hunk 2]
```diff
@@ -113,3 +113,45 @@ Tests should be run in the Docker container, not the host machine. They can be r
...
+### Running the app in a notebook
+
+This repo includes a config for running a [Jupyter notebook](https://jupyter.org/) in a
+Docker container. This enables you to do in a Jupyter notebook anything you might
+otherwise do in a Django shell. To get started:
+
+- Copy the example file
+    ```bash
+    # Choose any name for the resulting .ipynb file
+    cp localdev/app.ipynb.example localdev/app.ipynb
```
[the remark this answers is not in the document]
Comment (noisecapella) [reply {{ cookiecutter.project_name }}/README.md:65]: Thanks, just added it

File: {{ cookiecutter.project_name }}/localdev/app.ipynb.example [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    "ABS_PROJECT_ROOT_PATH = \"/src\"\n",
+    "BASE_DJANGO_APP_NAME = os.environ.get(\"BASE_DJANGO_APP_NAME\")\n",
+    "\n",
+    "if not ABS_PROJECT_ROOT_PATH in sys.path:\n",
+    "    sys.path.insert(0, ABS_PROJECT_ROOT_PATH)\n",
+    "os.environ.setdefault('DJANGO_SETTINGS_MODULE', '{}.settings'.format(BASE_DJANGO_APP_NAME))\n",
+    "django.setup()\n",
+    "\n",
+    "User = get_user_model()"
+   ]
```
Comment (mbertrand) [inline {{ cookiecutter.project_name }}/localdev/app.ipynb.example:23]: with the app now named 'main', the content should be modified to this (get rid of `BASE_DJANGO_APP_NAME`):
```python
import os
import sys
import django
from django.contrib.auth import get_user_model

ABS_PROJECT_ROOT_PATH = "/src"
if not ABS_PROJECT_ROOT_PATH in sys.path:
    sys.path.insert(0, ABS_PROJECT_ROOT_PATH)
os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'main.settings')
django.setup()

User = get_user_model()
```

**Pull request merged**